pyam/utils.py

```diff
@@ -585,3 +585,11 @@ def reduce_hierarchy(x, depth):
     _x = x.split('|')
     depth = len(_x) + depth - 1 if depth < 0 else depth
     return '|'.join(_x[0:(depth + 1)])
+
+
+def get_variable_components(x, depths):
```

Comment:
Suggest to rename the arg to `level` similar to [pandas.Index.get_level_values(level)](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Index.get_level_values.html)

Also, I think that the most common use case will to just separate the components and return as a list, so I would change the behavior to the following options:

```python
get_variable_components('foo|bar|baz, [0, 1])  # default returns list
> ['foo', 'bar']
get_variable_components('foo|bar|baz, [0, 1], join=True)  # True returns IAMC-style
> 'foo|bar'
get_variable_components('foo|bar|baz, [0, 1], join='-')  # any other symbol can be used to concatenate
> 'foo-bar'
```

```suggestion
def get_variable_components(x, level, join=False):
```

Reply:
Great suggestions! I will implement all ;-)